Reduce memory allocated in 'sp_context.c' used for caching. By default,
more than 100 MB of memory gets allocated via malloc. We override the
definitions in 'gallium/include/pipe/p_state.h' to reduce the footprint.

--- src/gallium/include/pipe/p_state.h	2010-04-05 18:05:15.000000000 +0200
+++ src/gallium/include/pipe/p_state.h	2010-08-07 22:45:24.000000000 +0200
@@ -56,10 +56,10 @@
  */
 #define PIPE_MAX_ATTRIBS          32
 #define PIPE_MAX_CLIP_PLANES       6
-#define PIPE_MAX_COLOR_BUFS        8
+#define PIPE_MAX_COLOR_BUFS        1 /* original value was 8 */
 #define PIPE_MAX_CONSTANT_BUFFERS 32
-#define PIPE_MAX_SAMPLERS         16
-#define PIPE_MAX_VERTEX_SAMPLERS  16
+#define PIPE_MAX_SAMPLERS          1 /* original value was 16 */
+#define PIPE_MAX_VERTEX_SAMPLERS   1 /* original value was 16 */
 #define PIPE_MAX_SHADER_INPUTS    16
 #define PIPE_MAX_SHADER_OUTPUTS   16
 #define PIPE_MAX_TEXTURE_LEVELS   16
